### PR TITLE
Use repeatble cache key from spl_object_id instead of spl_object_hash

### PR DESCRIPTION
**Pull request by mvorisek**
improved https://github.com/ondrejmirtes/BetterReflection/pull/9 , do not close PRs to be improved immediately!

### CODE REVIEW

File: src/SourceLocator/Type/MemoizingSourceLocator.php [hunk 1]
```diff
@@ -58,7 +59,7 @@ public function locateIdentifiersByType(Reflector $reflector, IdentifierType $id
     private function reflectorCacheKey(Reflector $reflector) : string
     {
         return 'type:' . get_class($reflector)
-            . '#oid:' . spl_object_hash($reflector);
+            . '#oid:' . (\PHP_MAJOR_VERSION > 7 || \PHP_MINOR_VERSION > 1 ? spl_object_id($reflector) : spl_object_hash($reflector));
```
Comment (ondrejmirtes) [inline src/SourceLocator/Type/MemoizingSourceLocator.php:62]: Can you just use function_exists?

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/SourceLocator/Type/MemoizingSourceLocator.php (hunk 1 above)
Comment (mvorisek) [reply src/SourceLocator/Type/MemoizingSourceLocator.php:62]: no - const can be resolved on compile time and the code should be dropped once PHP 7.1 support is dropped

**Comment by mvorisek** [comment]
FYI merged into the upstream

File: src/SourceLocator/Type/MemoizingSourceLocator.php (hunk 1 above)
Comment (ondrejmirtes) [reply src/SourceLocator/Type/MemoizingSourceLocator.php:64]: I don't care about "can be resolved on compile time". I care about that code being understood by PHPStan.

Comment (mvorisek) [reply src/SourceLocator/Type/MemoizingSourceLocator.php:64]: what is wrong with constants?

**Comment by mvorisek** [comment]
Merged in the upstream, not pursuing it here if there is no interest.

**Pull request closed without merging**